Race condition when CR and CRDs are part deployed together with Kustomize

Both watcher CR and CRD are deployed within the same Kustomize deployment 
https://github.com/kyma-project/lifecycle-manager/blob/main/operator/config/default/kustomization.yaml#L26

CRDs and CRs cannot be a part of the same Kustomize deploy, since order is not guaranteed OR there could be chance that the API server responds with a latency. Eventually it works with retries but a better strategy is required.

Reference: https://github.com/kubernetes-sigs/kustomize/issues/3502

```
customresourcedefinition.apiextensions.k8s.io/kymas.operator.kyma-project.io created
customresourcedefinition.apiextensions.k8s.io/moduletemplates.operator.kyma-project.io created
customresourcedefinition.apiextensions.k8s.io/watchers.operator.kyma-project.io created
serviceaccount/lifecycle-manager-controller-manager created
role.rbac.authorization.k8s.io/lifecycle-manager-leader-election-role created
clusterrole.rbac.authorization.k8s.io/lifecycle-manager-manager-role created
clusterrole.rbac.authorization.k8s.io/lifecycle-manager-manager-role-manifest created
clusterrole.rbac.authorization.k8s.io/lifecycle-manager-metrics-reader created
rolebinding.rbac.authorization.k8s.io/lifecycle-manager-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/lifecycle-manager-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/lifecycle-manager-manager-rolebinding-manifest created
clusterrolebinding.rbac.authorization.k8s.io/lifecycle-manager-metrics-rolebinding created
configmap/lifecycle-manager-manager-config created
service/lifecycle-manager-controller-manager-metrics-service created
service/lifecycle-manager-event-service created
deployment.apps/lifecycle-manager-controller-manager created
error: resource mapping not found for name: "lifecycle-manager-watcher" namespace: "kcp-system" from "STDIN": no matches for kind "Watcher" in version "operator.kyma-project.io/v1alpha1"
ensure CRDs are installed first
make: *** [deploy] Error 1
```

**Acceptance Criteria:**
- [ ] Introduce the retries in the Kustomize installation
- [ ] Have a flag for the number of retries (default to 3)
- [ ] Revisit the timeout flag (retries need the proper timeout configuration)

--- a/Makefile
+++ b/Makefile
@@ -116,7 +116,7 @@ uninstall: manifests kustomize ## Uninstall CRDs from the K8s cluster specified 
 	$(KUSTOMIZE) build config/crd | kubectl delete --ignore-not-found=$(ignore-not-found) -f -
 
 .PHONY: deploy
-deploy: manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+deploy: install manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
 	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 	

